substrate-node/pallets/pallet-tfgrid/src/lib.rs

```diff
@@ -92,12 +111,39 @@ pub mod pallet {
     pub type FarmPayoutV2AddressByFarmID<T: Config> =
         StorageMap<_, Blake2_128Concat, u32, Vec<u8>, ValueQuery>;
 
+    // Input type for public config
+    pub type PubConfigIP4Input = IP<
+        BoundedVec<u8, ConstU32<{ pub_config::MAX_IP_LENGTH }>>,
+        BoundedVec<u8, ConstU32<{ pub_config::MAX_GATEWAY_LENGTH }>>,
+    >;
+    pub type PubConfigIP6Input = IP<
+        BoundedVec<u8, ConstU32<{ pub_config::MAX_IP6_LENGTH }>>,
+        BoundedVec<u8, ConstU32<{ pub_config::MAX_IP6_LENGTH }>>,
+    >;
+    pub type PubConfigInput = PublicConfig<
+        PubConfigIP4Input,
```

Comment:
This probably needs some checking with what a public node means exactly in zos, but I imagine in the future IPv6 only pub nodes could exist, i.e. ipv4 optional? not sure though

Reply:
Let's leave this for next iteration?